File: Web/RMA-Web-Shell/projects/fincare/src/app/billing-manager/models/imports/bulk-manual-allocation.ts
import { dataExchangeModelBase } from "projects/shared-models-lib/src/lib/data-exchange/data-exchange-model";

export class BulkManualAllocation extends dataExchangeModelBase{
  public id: number;
  public bankAccountNumber: string;
  public userReference: string;
  public statementReference: string;
  public transactionDate: string; // staging TransactionDate as is from uploaded file
  public amount: string; // staging table Amount as is from uploaded file
  public status: string;
  public userReference2: string;
  public referenceType: string;
  public allocatable: string;
  public allocateTo: string;
  public bulkAllocationFileId: number;
  public error: string;
  public isDeleted: boolean;
  public lineProcessingStatusId?: number;
  public lineProcessingStatus: string;
  public periodId: number;
}

File: Service/Billing/RMA.Service.Billing.Contracts/Entities/BulkManualAllocation.cs
namespace RMA.Service.Billing.Contracts.Entities
{
    public class BulkManualAllocation
    {
        public int Id { get; set; } // Id (Primary key)
        public string BankAccountNumber { get; set; } // BankAccountNumber (length: 50)
        public string UserReference { get; set; } // UserReference (length: 200)
        public string StatementReference { get; set; } // StatementReference (length: 200)
        public string TransactionDate { get; set; } // TransactionDate (length: 50)
        public string Amount { get; set; } // Amount (length: 50)
        public string Status { get; set; } // Status (length: 50)
        public string UserReference2 { get; set; } // UserReference2 (length: 200)
        public string ReferenceType { get; set; } // ReferenceType (length: 100)
        public string Allocatable { get; set; } // Allocatable (length: 10)
        public string AllocateTo { get; set; } // AllocateTo (length: 200)
        public int BulkAllocationFileId { get; set; } // BulkAllocationFileId
        public string Error { get; set; } // Error (length: 500)
        public bool IsDeleted { get; set; } // IsDeleted
        public int? LineProcessingStatusId { get; set; } // LineProcessingStatusId
        public string LineProcessingStatus { get; set; } // Status (length: 50)
        public int? PeriodId { get; set; }
    }
}
